File: erlang/roman_numerals/test/roman_numerals_test.erl
-module(roman_numerals_test).

-compile(export_all).
-import(roman_numerals, [roman_of/1]).

 -include_lib("eunit/include/eunit.hrl").

there_is_no_roman_for_non_positive_numbers_test() ->
  ?assertEqual("", roman_of(0)),
  ?assertEqual("", roman_of(-1)).

number_to_roman_test() ->
  ?assertEqual("I", roman_of(1)),
  ?assertEqual("II", roman_of(2)),
  ?assertEqual("III", roman_of(3)),
  ?assertEqual("V", roman_of(5)),
  ?assertEqual("VII", roman_of(7)),
  ?assertEqual("X", roman_of(10)),
  ?assertEqual("XVI", roman_of(16)),
  ?assertEqual("XXX", roman_of(30)),
  ?assertEqual("IV", roman_of(4)),
  ?assertEqual("IX", roman_of(9)),
  
  ?assertEqual("MCMLXXXIV", roman_of(1984)),
  ?assertEqual("MMMCMXCIX", roman_of(3999)).

File: elixir/roman_numerals/test/roman_numerals_test.exs
defmodule RomanNumeralsTest do
  use ExUnit.Case
  use RomanNumerals
  doctest RomanNumerals

  test "no roman of numbers smaller than 1 or bigger than 3999" do
    assert roman_of(0) == ""
    assert roman_of(-1) == ""
    assert roman_of(4000) == ""
  end

  test "roman of" do
    assert roman_of(1) == "I" 
    assert roman_of(2) == "II" 
    assert roman_of(3) == "III" 
    assert roman_of(5) == "V" 
    assert roman_of(8) == "VIII" 
    assert roman_of(10) == "X" 
    assert roman_of(30) == "XXX" 
    assert roman_of(4) == "IV" 
    assert roman_of(9) == "IX" 
    assert roman_of(50) == "L" 
    assert roman_of(40) == "XL" 

    assert roman_of(1984) == "MCMLXXXIV" 
    assert roman_of(3999) == "MMMCMXCIX" 
  end
end
